### PR TITLE
Add default resource definition support for analyzers

### PR DESCRIPTION
**Pull request by sushicw**
Fixes https://github.com/istio/istio/issues/19579

Adds support for "default" resources to better support files-only use cases of `istioctl analyze`. This means that users only looking at files don't need to supply certain Istio-supplied resources in order to avoid false positives. We were previously solving this with hardcoded special cases in affected analyzers, this is a bit better. Currently we only need a service & pod for the default ingress gateway, more can be added if needed but this should be done extremely sparingly. 

Also make the analyzers_test output a bit easier to debug, and refactor how we handle mesh config slightly.

### CODE REVIEW

File: galley/pkg/config/analysis/local/analyze.go [hunk 1]
```diff
@@ -102,6 +106,7 @@ func NewSourceAnalyzer(m *schema.Metadata, analyzer *analysis.CombinedAnalyzer,
 
 	sa := &SourceAnalyzer{
 		m:                    m,
+		meshCfg:              meshcfg.Default(),
```
Comment (howardjohn) [inline galley/pkg/config/analysis/local/analyze.go]: is this still using some random default mesh config in galley code that isn't used by the rest of istio (namely pilot)? I think we changed it but not certain

File: galley/pkg/config/analysis/local/defaults.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  - name: tls
+    nodePort: 31447
+    port: 15443
+    protocol: TCP
+    targetPort: 15443
+  selector:
+    istio: ingressgateway
+`
+
+func getDefaultIstioIngress(namespace, ingressService string) (string, error) {
```
Comment (howardjohn) [inline galley/pkg/config/analysis/local/defaults.go]: confusing that this is not an Ingress, I suggest renaming to ingressgateway

File: galley/pkg/config/analysis/local/analyze.go (hunk 1 above)
Comment (sushicw) [reply galley/pkg/config/analysis/local/analyze.go]: The `meshcfg.Default()` being invoked here appears to only be used by Galley. 
https://github.com/istio/istio/blob/master/galley/pkg/config/meshcfg/defaults.go

It looks like the only difference between it and `mesh.DefaultMeshConfig` (which does appear to be used by Pilot as well) is that Galley's version has the ingress stuff turned on. 

If this is something we need to change, it should be handled outside the scope of this PR.

File: galley/pkg/config/analysis/local/defaults.go (hunk 2 above)
Comment (sushicw) [reply galley/pkg/config/analysis/local/defaults.go]: Done.

**Comment by howardjohn** [comment]
I think that part is fine, it used to be completely different

File: galley/pkg/config/analysis/local/defaults.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local
+
+import (
+	"bytes"
+	"html/template"
```
Comment (ayj) [inline galley/pkg/config/analysis/local/defaults.go]: `text/template`

Comment (sushicw) [reply galley/pkg/config/analysis/local/defaults.go]: Fixed

**Pull request merged**